libmachine/cert/cert.go

```diff
@@ -266,19 +266,19 @@ func (xcg *X509CertGenerator) ValidateCertificate(addr string, authOptions *auth
 		TLSClientConfig: tlsConfig,
 	}
 
-	envVar := util.GetProxyURL()
-	if envVar != "" {
-		log.Debugf("proxy address is used: %s", envVar)
-		url, err := url.Parse("http://" + envVar)
-		if err != nil {
-			return false, err
-		}
-		transport.Proxy = http.ProxyURL(url)
+	proxy, err := util.GetProxyURL(addr, "https")
```

Comment:
I think that just ensuring that the scheme is prefixed to `addr` is enough. The `ProxyFromEnvironment` function will do the correct thing.